src/config.c

```diff
@@ -0,0 +1,149 @@
+    // A value greater than 0.25 is treated as 0.25
+    .bf_error_rate =
+        {
+            .value = 0.01,
+            .min = 0.0,
+            .max = 1.0,
+        },
+    .bf_initial_size =
+        {
+            .value = 100,
+            .min = 1,
+            .max = 1LL << 30,
+        },
+    .bf_expansion_factor =
+        {
+            .value = 2,
+            .min = 0,
+            .max = 32768,
+        },
+    .cf_bucket_size =
+        {
+            .value = 2,
+            .min = 1,
+            .max = 255,
+        },
+    .cf_initial_size =
+        {
+            .value = 1024,
+            .min = 4,
+            .max = 1LL << 30,
+        },
+    .cf_max_iterations =
+        {
+            .value = 20,
+            .min = 1,
+            .max = 65535,
+        },
+    .cf_expansion_factor =
+        {
+            .value = 1,
+            .min = 0,
+            .max = 32768,
+        },
+    .cf_max_expansions =
+        {
+            .value = 32,
+            .min = 1,
+            .max = 65536,
+        },
+};
+
+static int setFloatValue(const char *name, RedisModuleString *value, void *privdata,
+                         RedisModuleString **err) {
+    RM_ConfigFloat *config = privdata;
+    double new_val;
+    if (RedisModule_StringToDouble(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (!isFloatConfigValid(new_val, *config)) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range (%f, %f)",
+                                              name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    config->value = new_val;
+    RedisModule_FreeString(NULL, config->str_value);
+    config->str_value = RedisModule_HoldString(NULL, value);
+    return REDISMODULE_OK;
+}
+static RedisModuleString *getFloatValue(const char *name, void *privdata) {
+    return ((RM_ConfigFloat *)privdata)->str_value;
+}
+
+static int setIntegerValue(const char *name, RedisModuleString *value, void *privdata,
+                           RedisModuleString **err) {
+    RM_ConfigInteger *config = privdata;
+    long long new_val;
+    if (RedisModule_StringToLongLong(value, &new_val) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(NULL, "Invalid value for `%s`", name);
+        return REDISMODULE_ERR;
+    }
+    if (!isIntegerConfigValid(new_val, *config)) {
+        *err =
+            RedisModule_CreateStringPrintf(NULL, "Value for `%s` must be in the range [%lld, %lld]",
+                                           name, config->min, config->max);
+        return REDISMODULE_ERR;
+    }
+    config->value = new_val;
+    RedisModule_FreeString(NULL, config->str_value);
+    config->str_value = RedisModule_HoldString(NULL, value);
+
+    if (!RM_ConfigStrCaseCmp(name, cf_bucket_size)) {
+        rm_config.cf_initial_size.min = new_val * 2;
```

Comment:
If the `new_val` is > `127`, the multiplication will make this higher than the defined maximum value for the configuration, I just wanted to check if this is correct for the `cf_initial_size` and the way it is used.

Reply:
`cf-initial-size`'s minimum may be greater than 255, that's fine. its minimum is `max(0, cf-bucket-size * 2)`, but since `cf-bucket-size` is bounded by `[1 .. 255]`, which is strictly greater than 0, we don't need to clamp.